pkg/executor/show_stats.go

```diff
@@ -513,30 +538,42 @@ func (e *ShowExec) fetchShowStatsHealthy(ctx context.Context) {
 		fieldFilter = e.Extractor.Field()
 		fieldPatternsLike = e.Extractor.FieldPatternLike()
 	}
+	tableInfoResult := do.InfoSchema().ListTablesWithSpecialAttribute(infoschemacontext.PartitionAttribute)
```

Comment:
since each show with build a map out 
<img width="1182" height="766" alt="image" src="https://github.com/user-attachments/assets/5fb4a137-396d-4605-9072-245a8cdb8fe7" />

i just wonder, why not just keep this inside schema itself, then when create/delete table, we just maintain this partition info map just like dbinfo/tableinfo/placementRule and so on.

Reply:
You may look at the old implementation. You should check the v2 implementation, and I think we mention it in the data structure.